test-one-file.lua

```diff
@@ -5,68 +5,218 @@
+}
+
+local dict39phn
+if opt.target == "phn" then
+   dict39phn = data.dictcollapsephones{dictionary=dict}
+   if opt.dict39 then
+      dict = dict39phn
+   end
+end
+
+if opt.dictsil then
+   data.dictadd{dictionary=dict, token='N', idx=assert(dict['|'])}
+   data.dictadd{dictionary=dict, token='L', idx=assert(dict['|'])}
+end
+
+if opt.ctc or opt.garbage then
+   data.dictadd{dictionary=dict, token="#"} -- blank
+end
+
+if opt.replabel > 0 then
+   for i=1,opt.replabel do
+      data.dictadd{dictionary=dict, token=string.format("%d", i)}
+   end
+end
+
+print(string.format('| number of classes (network) = %d', #dict))
+
+--reloading network
+print(string.format('| reloading model <%s>', reload))
+local model = serial.loadmodel{filename=reload, arch=true}
+local network = model.arch.network
+local transitions = model.arch.transitions
+local config = model.config
+local kw = model.config.kw
+local dw = model.config.dw
+assert(kw and dw, 'kw and dw could not be found in model archive')
+
+if opt.nolsm then
+   for i=network:size(),1,-1 do
+      if torch.typename(network:get(i)) == 'nn.LogSoftMax' then
+         print('! removing nn.LogSoftMax layer ' .. i)
+         network:remove(i)
+      end
+   end
+end
+assert(not (opt.addlsm and opt.nolsm))
+if opt.addlsm then
+   if opt.gpu then
+      network:insert(nn.LogSoftMax():cuda(), network:size())
+   else
+      network:add(nn.LogSoftMax())
+   end
+end
+print(network)
+
+-- make sure we do not apply aug on this
+opt.aug = false
+opt.shift = opt.shift or 0
 
 local criterion
+
 if opt.msc then
-   criterion = nn.MultiStateFullConnectCriterion(opt.nclass/opt.nstate, opt.nstate)
+   criterion = nn.MultiStateFullConnectCriterion(#dict/opt.nstate, opt.nstate)
 else
-   criterion = nn.Viterbi(opt.nclass)
+   criterion = (opt.ctc and nn.ConnectionistTemporalCriterion(#dict, nil)) or nn.Viterbi(#dict)
+end
+if not opt.ctc then
+   criterion.transitions:copy(transitions)
 end
-criterion.transitions:copy(transitions)
+
+if opt.shift > 0 then
+   print(string.format("| using shift scheme (shift=%d dshift=%d)", opt.shift, opt.dshift))
+   network = nn.ShiftNet(network, opt.shift)
+end
+
+local function tostring(tensor)
+    local str = {}
+    tensor:apply(
+       function(idx)
+          local letter = dict[idx]
+          assert(letter)
+          table.insert(str, letter)
+       end
+    )
+    return table.concat(str)
+ end
 
 -- basic info
-local fwav = sndfile.SndFile(testopt.wav)
+local fwav = sndfile.SndFile(opt.wav)
 local fwavinfo = fwav:info()
 print(string.format('| number of frames: %d (%6.2fs) [samplerate: %d channels: %d]',
                     fwavinfo.frames, fwavinfo.frames/fwavinfo.samplerate, fwavinfo.samplerate, fwavinfo.channels))
 fwav:close()
 
 -- note that the current setup expect the whole file as a ByteTensor
-local fwav = torch.DiskFile(testopt.wav):binary():quiet()
-local wav = torch.ByteTensor(fwav:readByte(2^30))
+local wav = audio.load(opt.wav)
```

Comment:
`sndfile` package should be able to do this. Adding `audio` dependency seems unnecessary.

Reply:
Cool, I will fix this over the weekend and create a separate pull request for a docker container. -- Cheers!

Reply:
```local wav = fwav:readFloat(fwav:info().frames)``` works here.